fury/ui/elements.py

```diff
@@ -3052,3 +3054,298 @@ def directory_click_callback(self, i_ren, _obj, listboxitem):
+        points_arr = vertices_from_actor(self.shape.actor)
+        rotation_matrix = np.array(
+            [[np.cos(angle), np.sin(angle), 0], [-np.sin(angle), np.cos(angle), 0], [0, 0, 1]])
+        new_points_arr = np.matmul(points_arr, rotation_matrix)
+        set_polydata_vertices(self.shape._polygonPolyData, new_points_arr)
+        self.shape.actor.GetMapper().Update()
+
+    def resize(self, size):
+        """Resize the UI.
+        """
+        if self.shape_type == "line":
+            hyp = np.hypot(size[0], size[1])
+            self.shape.resize((hyp, 3))
+            self.rotate(angle=np.arctan2(size[1], size[0]))
+
+        elif self.shape_type == "quad":
+            self.shape.resize(size)
+
+        elif self.shape_type == "circle":
+            hyp = np.hypot(size[0], size[1])
+            self.shape.outer_radius = hyp
+
+    def left_button_pressed(self, i_ren, _obj, shape):
+        mode = self.drawpanel.current_mode
+        if mode == "selection":
+            click_pos = np.array(i_ren.event.position)
+            self._drag_offset = click_pos - self.position
+            i_ren.event.abort()
+        elif mode == "delete":
+            self._scene.rm(self.shape.actor)
+            i_ren.force_render()
+        else:
+            self.drawpanel.left_button_pressed(i_ren, _obj, self.drawpanel)
+
+    def left_button_dragged(self, i_ren, _obj, shape):
+        if self.drawpanel.current_mode == "selection":
+            if self._drag_offset is not None:
+                click_position = np.array(i_ren.event.position)
+                new_position = click_position - self._drag_offset - self.drawpanel.canvas.position
+                self.drawpanel.canvas.update_element(self, new_position)
+            i_ren.force_render()
+        else:
+            self.drawpanel.left_button_dragged(i_ren, _obj, self.drawpanel)
+
+
+class DrawPanel(UI):
+    """The main Canvas(Panel2D) on which everything would be drawn.
+    """
+
+    def __init__(self, size=(400, 400), position=(0, 0), is_draggable=False):
+        """Init this UI element.
+
+        Parameters
+        ----------
+        size : (int, int)
+            Width and height in pixels of this UI component.
+        position : (float, float), optional
+            (x, y) in pixels.
+        is_draggable : bool, optional
+            Whether the background canvas will be draggble or not.
+        """
+        self.panel_size = size
+        super(DrawPanel, self).__init__(position)
+        self.is_draggable = is_draggable
+        self.iren = None
+        self.current_mode = None
+        self.shape_list = []
+
+    def _setup(self):
+        """Setup this UI component.
+
+        Create a Canvas(Panel2D).
+        """
+        self.canvas = Panel2D(size=self.panel_size)
+        self.canvas.background.on_left_mouse_button_pressed = self.left_button_pressed
+        self.canvas.background.on_left_mouse_button_dragged = self.left_button_dragged
+
+        mode_data = {
```

Comment:
this `mode_data`, should be a private variable (`self._mode_data`). Then you create a property `self.mode_data` (read-only first in this PR, but in the close future, the user can add a mode easily by adding it in the dictionary)

Reply:
can you note this somewhere as a future new feature